### PR TITLE
Adds events on show/hide of showon elements

### PR DESCRIPTION
**Pull request by regularlabs**
This PR adds events to the showing and hiding of shown elements.

So now you can trigger stuff when elements are dynamically shown or hidden, like:

```
document.addEventListener('joomla:showon-show', () => {
	const editor = document.querySelector('.CodeMirror');
	editor && editor.CodeMirror.refresh();
});
```

### CODE REVIEW

**Comment by Fedik** [comment]
Thanks, looks good now. Ready for testing.

**Comment by regularlabs** [comment]
Sucks that we have to wait at least another 3 months before we can use this in our extensions to work around issues in Joomla.
Which means I'll have to implement some kind of work-around, which defies the reason for needing this in core for me.

File: build/media_source/system/js/showon.es6.js [hunk 1]
```diff
@@ -199,8 +199,14 @@ class Showon {
       if (field.tagName !== 'option') {
         if (showfield) {
           field.classList.remove('hidden');
+          field.dispatchEvent(new CustomEvent('joomla:showon-show'), {
+            bubbles: true,
```
Comment (Fedik) [inline build/media_source/system/js/showon.es6.js:203]: I wanted to test, and noticed it has a bug, this correct:
```
new CustomEvent('joomla:showon-show', {
bubbles: true,
})
```

Comment (regularlabs) [reply build/media_source/system/js/showon.es6.js:203]: Oops... sorry

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 43ff9c6643235718bd154f3785662041fd3149a6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35994">issues.joomla.org/tracker/joomla-cms/35994</a>.</sub>

**Comment by Fedik** [comment]
For test create a field with showon, and add next js somewhere:
```
document.addEventListener('joomla:showon-show', () => {
 alert('Show!')
});
document.addEventListener('joomla:showon-hide', () => {
 alert('Hide!')
});
```

You should get an alert when an element is shows and when it hides.

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 43ff9c6643235718bd154f3785662041fd3149a6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35994">issues.joomla.org/tracker/joomla-cms/35994</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35994">issues.joomla.org/tracker/joomla-cms/35994</a>.</sub>

**Comment by regularlabs** [comment]
Thanks!

**Comment by pritam825** [comment]
I have tested this item :white_check_mark: successfully on 37db4fa680cc9856c9241f2f5fbc35525d6f51af<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35994">issues.joomla.org/tracker/joomla-cms/35994</a>.</sub>

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on 37db4fa680cc9856c9241f2f5fbc35525d6f51af<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35994">issues.joomla.org/tracker/joomla-cms/35994</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**